python/lsst/rubintv/production/aos.py

```diff
@@ -0,0 +1,219 @@
+            raise ValueError(f"Expected two exposureIds, got {exposureIds}")
+        return exposureIds
+
+    def _run_command(self, command):
+        """Run a command as a subprocess.
+
+        Runs the specified command as a subprocess, storing the process on the
+        class in a thread-safe manner. It logs the start time, process ID
+        (PID), and waits for the command to complete. If the command fails
+        (return code is non-zero), it logs an error. Finally, it logs the
+        completion time and duration of the command execution.
+
+        Parameters
+        ----------
+        command : `str`
+            The command to be executed.
+        """
+        start_time = time()
+        process = subprocess.Popen(command, shell=False)
+        with self.lock:
+            self.runningProcesses.append(process)
+        self.log.info(f"Process started with PID {process.pid}")
+        retcode = process.wait()
+        end_time = time()
+        duration = end_time - start_time
+        with self.lock:
+            self.runningProcesses.remove(process)
+        if retcode != 0:
+            self.log.error(f"Command failed with return code {retcode}")
+        self.log.info(f"Command completed in {duration:.2f} seconds with return code {retcode}")
+
+    def launchDonutProcessing(self, exposureBytes, doRegister=False):
+        """Launches the donut processing for a pair of donut exposures.
+
+        Parameters:
+        -----------
+        exposureBytes : bytes
+            The byte representation of the donut exposures, from redis.
+        doRegister : bool, optional
+            Add --register-dataset-types on the command line?
+
+        Notes:
+        ------
+        This method extracts the exposure IDs from the given byte
+        representation and launches the donut processing for the pair of donut
+        exposures. If the instrument is "LSSTComCamSim", it adjusts the
+        exposure IDs by adding 5000000000000 to account for the way this is
+        recorded in the butler. The command is executed in a separate thread,
+        and recorded as being in progress on the class.
+        """
+        expId1, expId2 = self._extractExposureIds(exposureBytes)
+        if self.instrument == "LSSTComCamSim":
+            # simulated exp ids are in the year 702X so add this manually, as
+            # OCS doesn't know about the fact the butler will add this on. This
+            # is only true for LSSTComCamSim though.
+            self.log.info("Adding 50000000000000 to exposure ids to adjust for simulated LSSTComCamSim data")
+            expId1 += 5000000000000
+            expId2 += 5000000000000
+        self.log.info(f"Launching donut processing for donut pair: {expId1, expId2}")
+        query = f"exposure in ({expId1},{expId2}) and instrument='{self.instrument}'"
+        command = [
+            # stop black messing this section up
+            # fmt: off
+            "pipetask", "run",
```

Comment:
Thanks, I'll add a TODO. Also though, there was talk about doing all the AOS stuff totally differently, so depending on timings, that will also render this somewhat moot I think. Also curious if this concerns you if this is only for the summit butler, but we can talk about this when we come to actually work on this sometime.

Reply:
Oh, also, I'm not sure if we'll ever need `-j189` for LSSTCam, as that would mean full focal plane donuts, _i.e._ pistoning the entire camera. It might happen here and there, but it'll be extremely rare.